### PR TITLE
Interface to use librascal potentials through driver.py

### PR DESCRIPTION
**Pull request by ceriottm**
Probably we should also move to driver.py similar python interfaces, such as QUIP, YAFF, etc.

### CODE REVIEW

**Comment by mahrossi** [comment]
> Probably we should also move to driver.py similar python interfaces, such as QUIP, YAFF, etc.

that is a good idea

**Comment by ceriottm** [comment]
> > Probably we should also move to driver.py similar python interfaces, such as QUIP, YAFF, etc.
> 
> that is a good idea

Shall we merge this and then open an issue? I would like to give the people who use those interfaces to have a say (got burned by the manual.pdf issue!)

File: ipi/engine/forcefields.py [hunk 1]
```diff
@@ -829,7 +829,7 @@ def __init__(
             )
         except ValueError:
             raise ValueError(
-                "ERROR: Reading sGDML model " + self.model + " file failed."
+                "ERROR: Reading sGDML model " + self.sGDML_model + " file failed."
```
[the remark this answers is not in the document]
Comment (ceriottm) [reply ipi/engine/forcefields.py:832]: there's plenty of these "old school string compositions" in the codebase and I think that ATM we are compatible with old python3, so I wouldn't introduce these unless we do this as a concerted modernization effort

[the remark this answers is not in the document]
Comment (Luthaf) [reply ipi/engine/forcefields.py:832]: 👍

**Comment by ceriottm** [comment]
Looks good to me. We need to add an example, which IMO really calls for some kind of zenodo/figshare scheme to fetch a JSON model.

**Comment by max-veit** [comment]
The interface has now been merged on the librascal side (see cosmo-epfl/librascal#307), so we should think about merging this soon as well!

File: examples/librascal/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Should get the model from librascal or somewhere. TBD.
```
Comment (max-veit) [inline examples/librascal/README.md:1]: er, let's expand this... how about using the Zundel example from librascal?

File: ipi/engine/forcefields.py [hunk 3]
```diff
@@ -855,7 +855,7 @@ def __init__(
...
```
Comment (max-veit) [inline ipi/engine/forcefields.py:858]: Is this at all related to the current PR?

File: drivers/f90/driver.f90 [hunk 4]
```diff
@@ -96,7 +96,9 @@ PROGRAM DRIVER
          IF (cmdbuffer == "-u") THEN ! flag for unix socket
             inet = 0
             ccmd = 0
-         ELSEIF (cmdbuffer == "-h") THEN ! read the hostname
+         ELSEIF (cmdbuffer == "-h") THEN ! read the hostname (deprecated)
+            ccmd = 1
+         ELSEIF (cmdbuffer == "-a") THEN ! read the hostname (address)
```
Comment (max-veit) [inline drivers/f90/driver.f90:101]: ...this also seems unrelated to the current PR and probably should be discussed separately (because it's changing the CLI of the driver!)

File: drivers/py/pes/harmonic.py [hunk 5]
```diff
@@ -13,7 +13,7 @@ def __init__(self, args=None):
     def check_arguments(self):
         """ Function that checks the arguments required to run the driver """
         try:
-            k = list(map(float, self.args.split()))
+            k = list(map(float, self.args.split(",")))
```
Comment (max-veit) [inline drivers/py/pes/harmonic.py:16]: also unrelated?

File: examples/librascal/BaTiO3_cubic_3x3x3.xyz [hunk 6]
```diff
@@ -0,0 +1,137 @@
+135
```
Comment (max-veit) [inline examples/librascal/BaTiO3_cubic_3x3x3.xyz:1]: Is this file used in any of the examples?

**Review by max-veit — COMMENTED** [review]
Just a few minor comments, the main thing is to improve the documentation (on the i-PI side, now). It helps that we already have an example for doing this on the librascal side.

File: drivers/py/pes/harmonic.py (hunk 5 above)
Comment (ceriottm) [reply drivers/py/pes/harmonic.py:16]: welcome to the far West Max.
here no random bugfix is left behind

File: drivers/f90/driver.f90 (hunk 4 above)
Comment (ceriottm) [reply drivers/f90/driver.f90:101]: it's related - argparse has special meaning for -h. This doesn't change (yet) the CLI of the fortran driver, also adds an "alias" so that -a works the same for both

**Comment by ceriottm** [comment]
ATM I'm wgetting from my dropbox. Not a long-term solution but we really need to merge this ASAP because there are a couple of other branches that need this - including one that is trying to address the problem of repo bloat.

File: examples/librascal/README.md (hunk 2 above)
Comment (ceriottm) [reply examples/librascal/README.md]: see latest commit.

File: ipi/engine/forcefields.py (hunk 3 above)
Comment (mahrossi) [reply ipi/engine/forcefields.py:858]: I think it is ok to leave it here but I agree this does not seem related to the current PR.

File: drivers/py/pes/rascal.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        pos_rascal = unit_to_user("length", "angstrom", pos)
+        cell_rascal = unit_to_user("length", "angstrom", cell)
+        # Do the actual calculation
+        pot, force, stress = self.rascal_calc.calculate(pos_rascal, cell_rascal)
+        pot_ipi = unit_to_internal("energy", "electronvolt", pot)
+        force_ipi = unit_to_internal("force", "ev/ang", force)
+        # The rascal stress is normalized by the cell volume (in rascal units)
+        vir_rascal = -1 * stress * det_ut3x3(cell_rascal)
+        vir_ipi = unit_to_internal("energy", "electronvolt", vir_rascal)
+        extras = "nada"
```
Comment (mahrossi) [inline drivers/py/pes/rascal.py:56]: why not pass a real empty string to extras, instead of writing something all the time?

**Review by mahrossi — APPROVED** [review]
looks good, only very small comments.
